### PR TITLE
feat(edit-draft-invoice): Apply adjusted fee on charge fee

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Currently it is not possible to manually edit the draft invoice through the UI. With this feature it will be possible to adjust fee, remove adjusted values and recalculate draft invoice based on those values.

## Description

Refresh draft invoice action can be called anytime, and this action destroys all attached fees and rebuilds them again. In a similar way also works finalise draft invoice action. This PR ensures that with each rebuilding of fees we check if there is any adjusted fee and we apply it on related fee. To be more specific, this PR covers rebuilding of charge fees if adjusted fee is preset

### CODE REVIEW

File: spec/services/fees/charge_service_spec.rb [hunk 1]
```diff
@@ -195,6 +195,264 @@
...
+              amount_currency: 'EUR',
+              units: 3,
+              unit_amount_cents: 2_000,
+              precise_unit_amount: 20,
+              events_count: 0,
+              payment_status: 'pending',
+            )
+          end
+
+          context 'and with true-up fee' do
```
Comment (jdenquin) [inline spec/services/fees/charge_service_spec.rb:243]: RSpec/ContextWording: Context description should match /^when\b/, /^with\b/, or /^without\b/. (https://rspec.rubystyle.guide/#context-descriptions, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording)

File: app/services/fees/charge_service.rb [hunk 2]
```diff
@@ -101,6 +121,75 @@ def init_fee(properties:, group: nil)
...
+    def adjusted_fee(group)
+      @adjusted_fee ||= {}
+
+      key = group ? group.id : 'default'
+
+      return @adjusted_fee[key] if @adjusted_fee.key?(key)
+
+      @adjusted_fee[key] = AdjustedFee
+        .where(invoice:, subscription:, charge:, group:, fee_type: :charge)
+        .where("(properties->>'charges_from_datetime')::date = ?", boundaries.charges_from_datetime.to_date)
```
Comment (lovrocolic) [inline app/services/fees/charge_service.rb:174]: switch from date to datetime

File: app/services/fees/charge_service.rb [hunk 3]
```diff
@@ -59,6 +68,17 @@ def init_fees
     end
 
     def init_fee(properties:, group: nil)
+      if invoice.draft? && adjusted_fee(group) && !adjusted_fee(group).adjusted_display_name?
```
Comment (vincent-pochet) [inline app/services/fees/charge_service.rb:71]: Could you add a comment, or extract the condition in a method, to help understand what we are doing here?

**Review by vincent-pochet — APPROVED** [review]
Small comments, but looks good!

**Review by ivannovosad — APPROVED** [review]

**Pull request merged**